index.js

```diff
@@ -59,30 +55,45 @@ function saveLastCampaign(opts) {
         options = merge(options, opts);
     }
 
-    // Parse the query string
-    data = querystring.parse(pageQueryString);
-
     // Set default cookie options
     cookieOptions = {
         domain: options.domain,
         path: options.path
     };
 
-    var removed = false;
+    // Parse the query string
+    if (pageQueryString.length !== 0) {
 
-    // Create the cookies
-    options.params.forEach(function (key) {
-        if (data[key]) {
+        data = querystring.parse(pageQueryString);
 
-            // param found in querystring so remove all necessary cookies first
-            if (!removed) {
-                removeCookies(options, cookieOptions);
-                removed = true;
+        var removed = false;
+
+        // Create the cookies
+        options.params.forEach(function (key) {
+            if (data[key]) {
+
+                // param found in querystring so remove all necessary existing cookies first
+                if (!removed) {
+                    removeCookies(options, cookieOptions);
+                    removed = true;
+                }
+
+                setCookie(cookie.serialize(options.prefix + key, data[key], cookieOptions));
             }
+        });
+    }
 
-            setCookie(cookie.serialize(options.prefix + key, data[key], cookieOptions));
+    // Save the data object
+    if (Object.keys(options.data).length !== 0) {
```

Comment:
[nit] `for ... in` will iterate over prototype chain.

You could reuse the array returned from Object.keys().

```
var dataKeys = Object.keys(options.data);
if (dataKeys.length == 0) {
    return;
}
dataKeys.forEach()
```